cypress/e2e/settings.cy.js

```diff
@@ -1,32 +1,85 @@
 /// <reference types='cypress' />
 /// <reference types='../support' />
+import SignInPageObject from '../support/pages/signIn.pageObject';
+import HomePageObject from '../support/pages/home.pageObject';
+import SettingsPageObject from '../support/pages/settings.pageObject';
+import faker from 'faker';
 
-describe('Settings page', () => {
-  before(() => {
+const signInPage = new SignInPageObject();
+const homePage = new HomePageObject();
+const settingsPage = new SettingsPageObject();
 
-  });
+describe('Settings page', () => {
+  let user;
 
   beforeEach(() => {
+    cy.task('db:clear');
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+
+      signInPage.visit();
+
+      cy.register(user.email, user.username, user.password);
+
+      signInPage.typeEmail(user.email);
+      signInPage.typePassword(user.password);
+      signInPage.clickSignInBtn();
 
+      homePage.clickSettings();
+    });
   });
 
   it('should provide an ability to update username', () => {
+    const editedUsername = faker.lorem.words(1);
+    settingsPage.usernameInput.clear().type(editedUsername);
+    settingsPage.submitSettings();
 
+    cy.contains('Update successful!');
```

Comment:
Avoid using hardcoded strings for checking alerts. The text of the alert can change, and it would be better to move these strings to constants or configuration files.

Reply:
done